CASC/PortfolioMode.cpp

```diff
@@ -224,80 +223,116 @@ bool PortfolioMode::performStrategy(Shell::Property* property)
-     extra_opts.push("bsd=on:fsd=on"); // subsumption demodulation
-     extra_opts.push("to=lpo");           // lpo
-   }
-
-   // If contains integers, rationals and reals
-   if(prop.props() & (Property::PR_HAS_INTEGERS | Property::PR_HAS_RATS | Property::PR_HAS_REALS)){
-
-    extra_opts.push("hsm=on");             // Sets a sensible set of Joe's arithmetic rules (TACAS-21) 
-    extra_opts.push("gve=force:asg=force:canc=force:ev=force:pum=on"); // More drastic set of rules
-    extra_opts.push("sos=theory:sstl=5");  // theory sos with non-default limit 
-    extra_opts.push("thsq=on");            // theory split queues, default
-    extra_opts.push("thsq=on:thsqd=16");   // theory split queues, other ratio
-   }
-   // If contains datatypes
-   if(prop.props() & Property::PR_HAS_DT_CONSTRUCTORS){
-     extra_opts.push("gtg=exists_all:ind=struct");
-     extra_opts.push("ind=struct:sik=one:indgen=on:indoct=on:drc=off");
-     extra_opts.push("ind=struct:sik=one:indgen=on");
-     extra_opts.push("ind=struct:sik=one:indoct=on");
-     extra_opts.push("ind=struct:sik=all:indmd=1");
-   }
-
-   // If in SMT-COMP mode try guessing the goal (and adding the twee trick!)
-   if(env.options->schedule() == Options::Schedule::SMTCOMP){
-    extra_opts.push("gtg=exists_all:tgt=full");
-   }
-   else{
-   // Don't try this in SMT-COMP mode as it requires a goal
-    extra_opts.push("slsq=on");
-    extra_opts.push("tgt=full");
-   }
+    // If contains integers, rationals and reals
+    if(prop.props() & (Property::PR_HAS_INTEGERS | Property::PR_HAS_RATS | Property::PR_HAS_REALS)){
+      extra_opts.push("hsm=on");             // Sets a sensible set of Joe's arithmetic rules (TACAS-21) 
+      extra_opts.push("gve=force:asg=force:canc=force:ev=force:pum=on"); // More drastic set of rules
+      extra_opts.push("sos=theory:sstl=5");  // theory sos with non-default limit 
+      extra_opts.push("thsq=on");            // theory split queues, default
+      extra_opts.push("thsq=on:thsqd=16");   // theory split queues, other ratio
+    }
+    // If contains datatypes
+    if(prop.props() & Property::PR_HAS_DT_CONSTRUCTORS){
+      extra_opts.push("gtg=exists_all:ind=struct");
+      extra_opts.push("ind=struct:sik=one:indgen=on:indoct=on:drc=off");
+      extra_opts.push("ind=struct:sik=one:indgen=on");
+      extra_opts.push("ind=struct:sik=one:indoct=on");
+      extra_opts.push("ind=struct:sik=all:indmd=1");
+    }
+
+    // If in SMT-COMP mode try guessing the goal (and adding the twee trick!)
+    if(env.options->schedule() == Options::Schedule::SMTCOMP){
+      extra_opts.push("gtg=exists_all:tgt=full");
+    }
+    else
+    {
+      // Don't try this in SMT-COMP mode as it requires a goal
+      extra_opts.push("slsq=on");
+      extra_opts.push("tgt=full");
+    }
 
+    // (almost) all of the above are integrated into SNAKE schedules
+    if (env.options->schedule() == Options::Schedule::SNAKE_TPTP_UNS) 
+    {
+      extra_opts.reset();
```

Comment:
Instead of adding them and resetting it, perhaps this should be a branch at the top level, perhaps with different functions? I'm thinking about readability... the logical flow of branching on options->schedule isn't clear here.

Reply:
A good point! Although I was thinking of the part of the code as a very temporary mess (with lower coding standards), now I maybe have an idea how to streamline things a bit... Will push a fix in a bit!